packages/lib/activity/service.ts

```diff
@@ -4,82 +4,95 @@ import { prisma } from "@formbricks/database";
+        id: attribute.id,
+        type: "attribute",
+        createdAt: attribute.createdAt,
+        updatedAt: attribute.updatedAt,
+        attributeLabel: attribute.attributeClass.name,
+        attributeValue: attribute.value,
+        actionLabel: null,
+        actionDescription: null,
+        actionType: null,
+        displaySurveyName: null,
+      }));
+      const unifiedDisplays: TActivityFeedItem[] = displays.map((display) => ({
+        id: display.id,
+        type: "display",
+        createdAt: display.createdAt,
+        updatedAt: display.updatedAt,
+        attributeLabel: null,
+        attributeValue: null,
+        actionLabel: null,
+        actionDescription: null,
+        actionType: null,
+        displaySurveyName: display.survey.name,
+      }));
+      const unifiedEvents: TActivityFeedItem[] = sessions.flatMap((session) =>
+        session.events.map((event) => ({
+          id: event.id,
+          type: "event",
+          createdAt: event.createdAt,
+          updatedAt: null,
+          attributeLabel: null,
+          attributeValue: null,
+          actionLabel: event.eventClass?.name || null,
+          actionDescription: event.eventClass?.description || null,
+          actionType: event.eventClass?.type || null,
+          displaySurveyName: null,
+        }))
+      );
 
-  const unifiedAttributes: TActivityFeedItem[] = attributes.map((attribute) => ({
-    id: attribute.id,
-    type: "attribute",
-    createdAt: attribute.createdAt,
-    updatedAt: attribute.updatedAt,
-    attributeLabel: attribute.attributeClass.name,
-    attributeValue: attribute.value,
-    actionLabel: null,
-    actionDescription: null,
-    actionType: null,
-    displaySurveyName: null,
-  }));
-  const unifiedDisplays: TActivityFeedItem[] = displays.map((display) => ({
-    id: display.id,
-    type: "display",
-    createdAt: display.createdAt,
-    updatedAt: display.updatedAt,
-    attributeLabel: null,
-    attributeValue: null,
-    actionLabel: null,
-    actionDescription: null,
-    actionType: null,
-    displaySurveyName: display.survey.name,
-  }));
-  const unifiedEvents: TActivityFeedItem[] = sessions.flatMap((session) =>
-    session.events.map((event) => ({
-      id: event.id,
-      type: "event",
-      createdAt: event.createdAt,
-      updatedAt: null,
-      attributeLabel: null,
-      attributeValue: null,
-      actionLabel: event.eventClass?.name || null,
-      actionDescription: event.eventClass?.description || null,
-      actionType: event.eventClass?.type || null,
-      displaySurveyName: null,
-    }))
-  );
+      const unifiedList: TActivityFeedItem[] = [...unifiedAttributes, ...unifiedDisplays, ...unifiedEvents];
 
-  const unifiedList: TActivityFeedItem[] = [...unifiedAttributes, ...unifiedDisplays, ...unifiedEvents];
+      return unifiedList;
+    },
+    [`getActivityTimeline-${personId}`],
+    { tags: [personCache.tag.byId(personId)], revalidate: SERVICES_REVALIDATION_INTERVAL }
```

Comment:
Okay this would be cool if someone could help with this. I think I get you mean here.

Rather than performing the joins to get the session, displays and attributes, we should call the services individually passing in the personId, is this correct?

Reply:
Right :-)